disasterinfosite/models.py

```diff
@@ -263,6 +271,40 @@ def default_display_name(sender, instance, *args, **kwargs):
         instance.display_name = instance.name
 
 
+# looks up a point in a set of rasters and returns the first value it finds
+# or None if there are no rasters or the point is not within any of them.
+# basic algorithm for this function taken from the django-raster project version 0.6 at
+# https://github.com/geodesign/django-raster/blob/master/raster/utils.py
+def rasterPointLookup(rasterCollection, lng, lat, band=0):
+    # if we have no data at all, then save time and return None immediately
+    if rasterCollection.objects.only("bbox").first() is None:
+        return None
+
+    collectionSRS = rasterCollection.objects.only("bbox").first().bbox.srs
+    pnt = OGRGeometry('POINT({0} {1})'.format(lng, lat), srs=collectionSRS)
+    results = []
+
+    # deferring the raster field will let us speed things up by doing boundary checks against the much faster-to-retrive bbox
+    for tile in rasterCollection.objects.defer("rast").all():
+        # only bother to check for data if we're within the bounds
+        bbox = OGRGeometry(tile.bbox.wkt, srs=collectionSRS)
```

Comment:
I am now wondering if there's a way to use PostGIS's very well optimised geographic searches on the collection of bboxes to speed this up by not even iterating through the tiles.  In theory we can ask PostGIS to pick a tile from the bbox field, and only look any further at that.  I don't know how to do this through Django's API, and I'm not sure it's worth the time, but this could be another thing to look at if point-in-raster lookups do prove annoyingly slow.

Reply:
OH HMMM that'd be cool.